shared/src/main/java/net/blay09/mods/waystones/api/IWaystoneTeleportContext.java

```diff
@@ -52,4 +54,16 @@ public interface IWaystoneTeleportContext {
     boolean playsEffect();
 
     void setPlaysEffect(boolean playsEffect);
+
+    default Predicate<? super ItemStack> getConsumeItemPredicate() {
+        return stack -> getWarpMode().consumesItem();
+    };
+
+    default void setConsumeItemPredicate(Predicate<? super ItemStack> predicate) { }
+
+    default BiPredicate<? super Entity, ? super IWaystone> getAllowTeleportPredicate() {
```

Comment:
I don't want to rush putting the allow teleport predicate in the API since, unlike consumesItem, it seems very specific to its current implementation details right now. If anything it should take the full teleport context, but even then I'd want to take a step back and evaluate where this predicate is currently used first, and whether it truly belongs in the context.